### PR TITLE
Don't limit stimulus:manifest:update to only .js files

### LINKED ISSUE #76: Don't limit stimulus:manifest:update to only .js files
rollup and webpack can perform transpiration of source in languages like coffeescript.  The following monkeypatch will enable `stimulus:manifest:update` to not only pick up sources that strictly end in `.js`, but also controllers that end in `.js.whatever`.  Perhaps it should be expanded further to support things like `.ts`?

```ruby
module Stimulus::Manifest
  def import_and_register_controller(controllers_path, controller_path)
    controller_path = controller_path.relative_path_from(controllers_path).to_s
    module_path = controller_path.split('.').first
    controller_class_name = module_path.camelize.gsub(/::/, "__")
    tag_name = module_path.remove(/_controller/).gsub(/_/, "-").gsub(/\//, "--")

    <<~JS

      import #{controller_class_name} from "./#{controller_path}"
      application.register("#{tag_name}", #{controller_class_name})
    JS
  end

  def extract_controllers_from(directory)
    (directory.children.select { |e| e.to_s =~ /_controller\.js(\.\w+)?$/ } +
      directory.children.select(&:directory?).collect { |d| extract_controllers_from(d) }
    ).flatten.sort
  end
end
```

See https://www.ruby2js.com/examples/rails/stimulus_rollup and https://www.ruby2js.com/examples/stimulus/ for a complete walkthrough.

P.S.  In any case, the regular expression `/_controller.js/` should escape the period, thus: `/_controller\.js/`.

### PR DESCRIPTION
**Pull request by rubys**
Fixes #76

### CODE REVIEW

**Comment by rubys** [comment]
Notes:

 * this will now accept any extension, be it `.ts`, `.jsx`, `.coffee`, whatever
 * extensions will now be explicit in the generated manifest, even for `.js` files.

**Comment by dhh** [comment]
This is looking good, but there's enough logic here now that we should have some tests.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
         bundle install --jobs 4 --retry 3
 
     - name: Test
-      run: bin/test
+      run: bundle exec rake
```
[the remark this answers is not in the document]
Comment (rubys) [reply .github/workflows/ci.yml:33]: I'm flying blind here... I can't see the output of action runs.  What I do know is that there is no `bin/test` file [in this repository](https://github.com/hotwired/stimulus-rails/tree/main/bin), and that other ci setups in the set of rails repositories run `bundle exec rake` directly - [example]( https://github.com/rails/jbuilder/blob/master/.github/workflows/ruby.yml).  Perhaps at some point in the future there may be a need for something more involved than running rake directly, and perhaps a bin/test can be added at that time (YAGNI?)

[the remark this answers is not in the document]
Comment (dhh) [reply .github/workflows/ci.yml:33]: Ha, lol, must have been a copy'paste setup. All good then.

**Pull request merged**